src/modules/fancyzones/lib/FancyZones.cpp

```diff
@@ -323,36 +324,59 @@ FancyZones::VirtualDesktopInitialize() noexcept
     PostMessage(m_window, WM_PRIV_VD_INIT, 0, 0);
 }
 
+bool FancyZones::MoveWindowIntoLastKnownZone(HWND window, winrt::com_ptr<IZoneWindow> zoneWindow) {
+    const auto activeZoneSet = zoneWindow->ActiveZoneSet();
+    if (activeZoneSet)
+    {
+        wil::unique_cotaskmem_string guidString;
+        if (SUCCEEDED(StringFromCLSID(activeZoneSet->Id(), &guidString)))
+        {
+            auto& fancyZonesData = JSONHelpers::FancyZonesDataInstance();
+            std::vector<int> zoneIndexSet = fancyZonesData.GetAppLastZoneIndexSet(window, zoneWindow->UniqueId(), guidString.get());
+            if (zoneIndexSet.size() &&
+                !fancyZonesData.IsAnotherWindowOfApplicationInstanceZoned(window, zoneWindow->UniqueId()))
+            {
+                m_windowMoveHandler.MoveWindowIntoZoneByIndexSet(window, zoneIndexSet, zoneWindow);
+                fancyZonesData.UpdateProcessIdToHandleMap(window, zoneWindow->UniqueId());
+                return true;
+            }
+        }
+    }
+    return false;
+}
+
 // IFancyZonesCallback
 IFACEMETHODIMP_(void)
 FancyZones::WindowCreated(HWND window) noexcept
 {
-    std::shared_lock readLock(m_lock);
+    // Avoid processing splash screens and already stamped (zoned) windows.
+    if (IsSplashScreen(window) &&

```

Comment:
`&&` is not correct, should be `||`

Reply:
Also, let's do all the checks to stop processing the current window in one place, instead of having to call later
```
if (m_settings->GetSettings()->appLastZone_moveWindows &&
        IsInterestingWindow(window, m_settings->GetSettings()->excludedAppsArray))
```

Reply:
Done.